editor/src/components/inspector/flex-section.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { useEditorState } from '../editor/store/store-hook'
+import { FlexDirectionToggle } from './flex-direction-control'
+import { detectFlexDirection } from './inspector-common'
+import { NineBlockControl } from './nine-block-controls'
+
+export const FlexSection = React.memo(() => {
+  const metadata = useEditorState((store) => store.editor.jsxMetadata, 'FlexSection metadata')
```

Comment:
store.editor.jsxMetadata is expected to change almost every frame during an interaction, so if your selector returns that, it will trigger a re-render of FlexSection for every frame.

instead, it is better to move detectFlexDirection _inside_ the selector. then useEditorState's built-in shallow equality check will compare the previous flexDirection with the current flexDirection, which means it will only trigger a re-render if flexDirection _actually_ changes. 

it's also important to keep in mind that the selector needs to be very very fast, if you have to make a slow selector, there's tools to mitigate that like reselect, but the first approximation is that selectors should almost take no time to run!

Reply:
fixed!